libnmstate/nm/context.py

```diff
@@ -0,0 +1,73 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+import logging
+
+from .common import NM
+from .common import GLib
+
+
+class NmContext:
+    def __init__(self):
+        self._client = NM.Client.new(cancellable=None)
+        self._context = self._client.get_main_context()
+        self._quiting = False
+
+    @property
+    def client(self):
+        if self._quiting:
+            return None
+        return self._client
+
+    @property
+    def context(self):
+        return self._context
+
+    def refresh_content(self):
+        if self.context:
+            while self.context.iteration(False):
+                pass
+
+    def clean_up(self):
+        self._del_client()
+
+    def _del_client(self):
```

Comment:
This `NmContext` is not just holding `NM.Client`.

It will hold `MainContext` used by PR #1016 

I don't like `close`, `destroy` or `delete` name for this function.

Reply:
I do not like the current one, as it is not only deleting the client, but doing other stuff as well (finalizing the whole object).
But it is not worth blocking the entire thing.